com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/MyLanguageServerImpl.java

```diff
@@ -122,27 +122,25 @@ public void initialized(@Nullable InitializedParams params) {
    */
   void retrieveAndStoreConfiguration() {
     fetchSettings(LSP_PREFIX.label + "." + CPY_MANAGER.label, null)
-        .thenAccept(
-            e -> {
-              JsonObject jsonObject = (JsonObject) e.get(0);
-              ConfigurationSettingsStorable configurationSettingsStorable =
-                  parseJsonIfValid(jsonObject);
-              ((SettingsProvider) settingsProvider).set(configurationSettingsStorable);
-            });
+        .thenAccept(e -> getSettingsFromProvider().set(parseJsonIfValid((JsonObject) e.get(0))));
+  }
+
+  private SettingsProvider getSettingsFromProvider() {
+    return settingsProvider.get();
   }
 
   /**
    * @param jsonObject - the object which comes from the client and contains configuration settings
-   * @return a custom object of type ConfigurableSettingsStorage if the JSON is valid or null if it
-   *     is failing the check
+   * @return a custom object of type ConfigurableSettingsStorable if the JSON is valid or an empty
+   *     ConfigurableSettingsStorable if it is failing the parsing
    */
   private ConfigurationSettingsStorable parseJsonIfValid(JsonObject jsonObject) {
     Gson gson = new Gson();
     try {
       return gson.fromJson(jsonObject, ConfigurationSettingsStorable.class);
     } catch (JsonSyntaxException e) {
       log.error(e.getMessage());
-      return null;
+      return ConfigurationSettingsStorable.builder().build();
```

Comment:
if there was a problem with the JSON then why are you building the settings?

Reply:
because we need to avoid NPE on Copybook service when I ask the provider to get me the object. I designed a proper logic on CopybookService to check if the configuration are not valid.